ndarray/src/main/java/org/tensorflow/ndarray/Shape.java

```diff
@@ -177,6 +188,24 @@ public boolean isUnknown() {
     }
   }
 
+  /**
+   * Returns a defensive copy of the this Shape's axes. Changes to the returned list do not change this Shape's state.
+   * Returns null if {@link Shape#isUnknown()} is true.
+   */
+  public List<Long> toListOrNull() {
+    long[] array = asArray();
+    if (array == null) {
+      return null;
+    }
+
+    List<Long> list = new ArrayList<>(array.length);
+    for (long l : array) {
+      list.add(l);
+    }
+
+    return list;
```

Comment:
Should this list be immutable?

Reply:
It could be, but since it's a new list each time I didn't see any reason to force it.